tx-indexer/src/collector.rs

```diff
@@ -292,81 +297,83 @@ async fn save_transaction_details(
         }
     };
     let transaction_hash = transaction_details.transaction.hash.to_string();
-    let signer_id = transaction_details.transaction.signer_id.to_string();
-    let tx_bytes = &borsh::to_vec(&transaction_details).expect("Failed to serialize transaction");
+    let tx_bytes = borsh::to_vec(&transaction_details).expect("Failed to serialize transaction");
 
-    match db_manager
-        .add_transaction(
-            &transaction_hash,
-            tx_bytes.clone(),
-            tx_details.block_height,
-            &signer_id,
-        )
-        .await
-    {
-        Ok(_) => {
-            // Validate that the transaction is saved correctly
-            // Before validating, we need to wait a little bit to make sure that the transaction is saved
-            tokio::time::sleep(std::time::Duration::from_millis(150)).await;
-            let mut attempts = 1;
-            let mut tx_bytes = tx_bytes.clone();
-            while let Err(err) = db_manager
-                .validate_saved_transaction_deserializable(&transaction_hash, &tx_bytes)
-                .await
-            {
-                tracing::warn!(
+    // We faced the issue when the transaction was saved to the storage but later
+    // was failing to deserialize. To avoid this issue and monitor the situation
+    // without runing the user experience, we will try to save the transaction
+    // to the storage and validate that it is saved correctly for 3 attempts
+    // before we throw an error.
+    let mut save_attempts = 0;
+    'retry: loop {
+        save_attempts += 1;
+        if save_attempts > 3 {
```

Comment:
Let's increase the number of attempts to 10 to make sure we've done everything we can to save the transaction.